gdnative-core/src/core_types/dictionary.rs

```diff
@@ -69,21 +69,42 @@ impl<Access: ThreadAccess> Dictionary<Access> {
         unsafe { (get_api().godot_dictionary_has_all)(self.sys(), keys.sys()) }
     }
 
-    /// Returns a copy of the value corresponding to the key.
+    /// Returns a copy of the value corresponding to the key if it exists.
     #[inline]
-    pub fn get<K>(&self, key: K) -> Variant
+    pub fn get<K>(&self, key: K) -> Option<Variant>
     where
         K: ToVariant + ToVariantEq,
+    {
+        let key = key.to_variant();
+        self.contains(&key).then(|| self.get_or_nil(key))
+    }
+
+    /// Returns a copy of the value corresponding to the key, or `default` if it doesn't exist
+    #[inline]
+    pub fn get_or<K, D>(&self, default: D, key: K) -> Variant
```

Comment:
Why is `default` the first parameter? Having `key` as first parameter for all 3 `get*` methods is more intuitive.

Reply:
I was thinking of it in terms of `Option::map_or`/`Option::map_or_else`... Which admittedly aren't necessarily intuitive; I still have to think about them a bit when I use them. And I suppose this is a case where staying closer to GDScript is probably a good thing.